examples/matRad_publishExamples.m

```diff
@@ -0,0 +1,64 @@
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% matRad publish examples script
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2017 the matRad development team. 
+% 
+% This file is part of the matRad project. It is subject to the license 
+% terms in the LICENSE file found in the top-level directory of this 
+% distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part 
+% of the matRad project, including this file, may be copied, modified, 
+% propagated, or distributed except according to the terms contained in the 
+% LICENSE file.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%% This file will publish all examples
+
+
+%% clear environment
+clc,clear,close all;
+
+% make sure you are in the matRad root directory and have added the
+% exampels folder to the matlab path
+
+format     = {'pdf'}; %{'html','pdf'};
+matRadRoot = pwd;
+folderInfo = dir([matRadRoot filesep 'examples']);
+
+   
+for ixFile = 1:size(folderInfo,1)
+      
+    for ixFormat = 1:numel(format)
+        
+        %make sure not to use auto saved files (*.asv)
+        [pathstr,name,ext] = fileparts(folderInfo(ixFile).name);
+        
+       if ~folderInfo(ixFile).isdir && ~strcmp(name,mfilename) && strcmp(ext,'.m') ...
+           
+           save('workspace')
+           publish(folderInfo(ixFile).name,format{ixFormat});
+           clc,clear,close all
+           load('workspace')
```

Comment:
Why do we have the loading and saving of the workspace here?

Reply:
I had the problem that once the publish command has finished it automaticallyl deletes the base worksapce

Reply:
This is strange... are you sure there was no 'clear' command in the script you were running with 'publish'? Cannot find anything about it with google!

Reply:
![1](https://user-images.githubusercontent.com/1581106/31267436-6ad33b34-aa78-11e7-8ce8-c3024c2d7d5a.png)
Line 43 has not yet been executed - on the right side you see the base workspace which contains solely variables that were used inside the publishing function. 
![2](https://user-images.githubusercontent.com/1581106/31267492-a22edc00-aa78-11e7-990f-83e53049ac76.png)
The workspace was cleared and the original variables were loaded to properly continue with the loop.

Reply:
hmm I don't see my previous message anymore: Therefore I post it once again:
I had the problem that once the publish command finished you end up with the variables that were used in the publish function - (therefore ->clear all) The variables of matRad_publishExamples, that were available before the publish command are gone afterwards. It seems that matlab  internally deletes the base workspace. I could in prinicple use different workspaces but I thought just saving and loading the workspace is easier xD